File: leetcode/3541. 找到频率最高的元音和辅音.java
class Solution {
    public int maxFreqSum(String s) {
        // 用于统计元音字母频率的 map
        Map<Character, Integer> vowelMap = new HashMap<>();
        // 用于统计辅音字母频率的 map
        Map<Character, Integer> consonantMap = new HashMap<>();

        // 定义元音字母
        String vowels = "aeiou";

        // 遍历字符串中的每个字符
        for (char c : s.toCharArray()) {
            if (vowels.indexOf(c) != -1) {
                // 是元音
                vowelMap.put(c, vowelMap.getOrDefault(c, 0) + 1);
            } else {
                // 是辅音
                consonantMap.put(c, consonantMap.getOrDefault(c, 0) + 1);
            }
        }

        // 找出元音中出现频率最高的次数
        int maxVowelFreq = 0;
        for (int freq : vowelMap.values()) {
            maxVowelFreq = Math.max(maxVowelFreq, freq);
        }

        // 找出辅音中出现频率最高的次数
        int maxConsonantFreq = 0;
        for (int freq : consonantMap.values()) {
            maxConsonantFreq = Math.max(maxConsonantFreq, freq);
        }

        // 返回两者最大频率之和
        return maxVowelFreq + maxConsonantFreq;
    }
}

File: leetcode/3541. 找到频率最高的元音和辅音.rs
impl Solution {
    pub fn max_freq_sum(s: String) -> i32 {
        use std::collections::HashMap;
        
        let mut vowel_map: HashMap<char, i32> = HashMap::new();
        let mut consonant_map: HashMap<char, i32> = HashMap::new();
        
        // 定义元音字母集合（小写）
        let vowels = ['a', 'e', 'i', 'o', 'u'];
        
        for c in s.chars() {
            if vowels.contains(&c) {
                *vowel_map.entry(c).or_insert(0) += 1;
            } else {
                *consonant_map.entry(c).or_insert(0) += 1;
            }
        }
        
        // 找出元音中出现最多的频率
        let max_vowel_freq = vowel_map.into_iter().map(|(_, freq)| freq).max().unwrap_or(0);
        
        // 找出辅音中出现最多的频率
        let max_consonant_freq = consonant_map.into_iter().map(|(_, freq)| freq).max().unwrap_or(0);
        
        max_vowel_freq + max_consonant_freq
    }
}